cmd/argocd/commands/app.go

```diff
@@ -1506,6 +1506,8 @@ func setParameterOverrides(app *argoappv1.Application, parameters []string) {
 			sourceType = argoappv1.ApplicationSourceTypeKsonnet
 		} else if len(strings.SplitN(parameters[0], "=", 2)) == 2 {
 			sourceType = argoappv1.ApplicationSourceTypeHelm
+		} else {
+			sourceType = argoappv1.ApplicationSourceTypeKustomize
```

Comment:
I don't think we should overload the `-p` option in the CLI for this, and instead use a new kustomize specific flag. e.g.

```
agocd app set --kustomize-image nginx:1.2.3
```

Reply:
I was wondering if we should have a command like kustomize has, `kustomize edit set image` we could have something like `argocd app image ...`?

Reply:
@alexmt @jessesuen - how would you define parameters?